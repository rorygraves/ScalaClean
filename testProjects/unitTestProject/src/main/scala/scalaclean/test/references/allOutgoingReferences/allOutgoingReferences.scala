package scalaclean.test.references.allOutgoingReferences

class Parent {
  def foo: Unit = ()
}

class Child extends Parent {
  def bar(x: Any): Unit = ()
}

object Special {
  def myVar1           = 1
  def myVar1_=(i: Int) = ()

  def myVar2           = 1
  def myVar2_=(i: Int) = ()

  def apply()         = 7
  def unapply(a: Any) = Some(1, 2)

  def update(i: Int, j: Int) = ()

}

object XX {
  new Child().foo
  new Child().bar(1)

  Special.myVar1 += 1

  Special.myVar2 = 9
  println(Special.myVar2)

  Special()
  val x: Any = 1
  x match {
    case i: Int        => ???
    case Special(a, b) => ???
  }

  Special(1) = 7
}
