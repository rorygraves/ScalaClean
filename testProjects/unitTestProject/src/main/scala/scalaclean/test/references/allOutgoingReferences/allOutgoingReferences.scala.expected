package scalaclean.test.references.allOutgoingReferences

class Parent {
  def foo: Unit = ()/* allOutgoingReferences(C:scalaclean.test.references.allOutgoingReferences.Parent/M:foo()) - C:scala.Unit :: P:scala */
}/* allOutgoingReferences(C:scalaclean.test.references.allOutgoingReferences.Parent) - <root> :: C:java.lang.Object :: P:scala :: P:scalaT:AnyRef :: P:scalaclean.test.references.allOutgoingReferences */

class Child extends Parent {
  def bar(x: Any): Unit = ()/* allOutgoingReferences(C:scalaclean.test.references.allOutgoingReferences.Child/M:bar(scala.Any)) - C:scala.Any :: C:scala.Unit :: P:scala */
}/* allOutgoingReferences(C:scalaclean.test.references.allOutgoingReferences.Child) - C:scalaclean.test.references.allOutgoingReferences.Parent :: P:scalaclean.test.references.allOutgoingReferences */

object Special {
  def myVar1           = 1/* allOutgoingReferences(O:scalaclean.test.references.allOutgoingReferences.Special/M:myVar1()) - C:scala.Int :: P:scala */
  def myVar1_=(i: Int) = ()/* allOutgoingReferences(O:scalaclean.test.references.allOutgoingReferences.Special/M:myVar1_$eq(scala.Int)) - C:scala.Int :: C:scala.Unit :: P:scala */

  def myVar2           = 1/* allOutgoingReferences(O:scalaclean.test.references.allOutgoingReferences.Special/M:myVar2()) - C:scala.Int :: P:scala */
  def myVar2_=(i: Int) = ()/* allOutgoingReferences(O:scalaclean.test.references.allOutgoingReferences.Special/M:myVar2_$eq(scala.Int)) - C:scala.Int :: C:scala.Unit :: P:scala */

  def apply()         = 7/* allOutgoingReferences(O:scalaclean.test.references.allOutgoingReferences.Special/M:apply()) - C:scala.Int :: P:scala */
  def unapply(a: Any) = Some(1, 2)/* allOutgoingReferences(O:scalaclean.test.references.allOutgoingReferences.Special/M:unapply(scala.Any)) - <root> :: C:scala.Any :: C:scala.Int :: C:scala.Nothing :: C:scala.Some :: C:scala.Tuple2 :: O:scala.Some :: O:scala.Some :: O:scala.Some/M:apply[<none>](scala.Some.A) :: O:scala.Some/M:apply[<none>](scala.Some.A) :: O:scala.Tuple2 :: O:scala.Tuple2 :: O:scala.Tuple2/M:apply[<none>;<none>](scala.Tuple2.T1;scala.Tuple2.T2) :: O:scala.Tuple2/M:apply[<none>;<none>](scala.Tuple2.T1;scala.Tuple2.T2) :: P:scala */

  def update(i: Int, j: Int) = ()/* allOutgoingReferences(O:scalaclean.test.references.allOutgoingReferences.Special/M:update(scala.Int;scala.Int)) - C:scala.Int :: C:scala.Unit :: P:scala */

}/* allOutgoingReferences(O:scalaclean.test.references.allOutgoingReferences.Special) - <root> :: C:java.lang.Object :: C:java.lang.Object/M:<init>() :: C:scala.Unit :: P:java.lang :: P:scala :: P:scalaT:AnyRef :: P:scalaclean.test.references.allOutgoingReferences */

object XX {
  new Child().foo
  new Child().bar(1)

  Special.myVar1 += 1

  Special.myVar2 = 9
  println(Special.myVar2)

  Special()
  val x: Any = 1/* allOutgoingReferences(O:scalaclean.test.references.allOutgoingReferences.XX/V:x ) - C:scala.Any :: P:scala */
  x match {
    case i: Int        => ???
    case Special(a, b) => ???
  }

  Special(1) = 7
}/* allOutgoingReferences(O:scalaclean.test.references.allOutgoingReferences.XX) - <root> :: C:java.lang.Object :: C:java.lang.Object/M:<init>() :: C:scala.Any :: C:scala.Any :: C:scala.Any/M:isInstanceOf[<none>]() :: C:scala.Boolean :: C:scala.Boolean/M:unary_$bang() :: C:scala.Int :: C:scala.Int/M:$plus(scala.Int) :: C:scala.MatchError :: C:scala.MatchError/M:<init>(scala.Any) :: C:scala.Nothing :: C:scala.Some :: C:scala.Some :: C:scala.Some/M:isEmpty() :: C:scala.Tuple2 :: C:scala.Unit :: C:scalaclean.test.references.allOutgoingReferences.Child :: C:scalaclean.test.references.allOutgoingReferences.Child/M:<init>() :: C:scalaclean.test.references.allOutgoingReferences.Child/M:bar(scala.Any) :: C:scalaclean.test.references.allOutgoingReferences.Parent/M:foo() :: O:scala.Predef :: O:scala.Predef/M:$qmark$qmark$qmark() :: O:scala.Predef/M:println(scala.Any) :: O:scalaclean.test.references.allOutgoingReferences.Special :: O:scalaclean.test.references.allOutgoingReferences.Special/M:apply() :: O:scalaclean.test.references.allOutgoingReferences.Special/M:myVar1() :: O:scalaclean.test.references.allOutgoingReferences.Special/M:myVar1_$eq(scala.Int) :: O:scalaclean.test.references.allOutgoingReferences.Special/M:myVar2() :: O:scalaclean.test.references.allOutgoingReferences.Special/M:myVar2_$eq(scala.Int) :: O:scalaclean.test.references.allOutgoingReferences.Special/M:unapply(scala.Any) :: O:scalaclean.test.references.allOutgoingReferences.Special/M:update(scala.Int;scala.Int) :: O:scalaclean.test.references.allOutgoingReferences.XX/M:x() :: P:java.lang :: P:scala :: P:scalaT:AnyRef :: P:scalaclean.test.references.allOutgoingReferences */
